modules/combined/test/tests/optimization/invOpt_mechanics/tests

```diff
@@ -11,5 +11,17 @@
     max_threads = 1
     # steady solve
     recover = false
+    detail = 'using a Hessian based optimimization,'
+  []
+  [both_disp]
+    type = CSVDiff
+    rel_err = 1e-6
+    abs_zero = 1e-8
```

Comment:
This is still failing on one of the tests.  I think with these Tao settings:
```  
petsc_options_iname = '-tao_gttol -tao_max_it'
petsc_options_value = '1e-7 500 `
```
You will never be guaranteed to have a gradient below 1e-7.  You should set abs_zero to 1e-6.

Reply:
We probably have a lot of tests that are using a large value for abs_zero, like 1, to ignore the gradient.  We should be more careful about that and set abs_zero an order of magnitude larger than the tao solver tolerance to make sure it converged.